velox/type/Filter.cpp

```diff
@@ -302,6 +305,69 @@ bool BigintValuesUsingHashTable::testInt64Range(
   return max >= *it;
 }
 
+Int128ValuesUsingHashTable::Int128ValuesUsingHashTable(
+    int128_t min,
+    int128_t max,
+    const std::vector<int128_t>& values,
+    bool nullAllowed)
+    : Filter(true, nullAllowed, FilterKind::kBigintValuesUsingHashTable),
+      min_(min),
+      max_(max),
+      values_(values) {
+  constexpr int32_t kPaddingElements = 4;
+  VELOX_CHECK(min <= max, "min must be less than max");
+  VELOX_CHECK(!values.empty(), "values must not be empty");
+
+  // Size the hash table to be 2+x the entry count, e.g. 10 entries
+  // gets 1 << log2 of 50 == 32. The filter is expected to fail often so we
+  // wish to increase the chance of hitting empty on first probe.
+  auto size = 1u << (uint32_t)std::log2(values.size() * 5);
+  hashTable_.resize(size + kPaddingElements);
+  sizeMask_ = size - 1;
+  std::fill(hashTable_.begin(), hashTable_.end(), kEmptyMarker);
+  for (auto value : values) {
+    if (value == kEmptyMarker) {
+      containsEmptyMarker_ = true;
+    } else {
+      uint32_t position = ((value * M) & sizeMask_);
+      for (auto i = position; i < position + size; i++) {
+        uint32_t index = i & sizeMask_;
+        if (hashTable_[index] == kEmptyMarker) {
+          hashTable_[index] = value;
+          break;
+        }
+      }
+    }
+  }
+  // Replicate the last element of hashTable kPaddingEntries times at 'size_' so
+  // that one can load a full vector of elements past the last used index.
+  for (auto i = 0; i < kPaddingElements; ++i) {
+    hashTable_[sizeMask_ + 1 + i] = hashTable_[sizeMask_];
+  }
+  std::sort(values_.begin(), values_.end());
+}
+
+bool Int128ValuesUsingHashTable::testInt128(int128_t value) const {
+  if (containsEmptyMarker_ && value == kEmptyMarker) {
+    return true;
+  }
+  if (value < min_ || value > max_) {
+    return false;
+  }
+  uint32_t pos = (value * M) & sizeMask_;
```

Comment:
This is a 64-bit value, is it ok to use for 128-bit values as well. See if there is an 128-bit equivalent for murmur hash.
```
 // from Murmur hash
  static constexpr uint64_t M = 0xc6a4a7935bd1e995L;
```

Reply:
Thanks for the suggestion, the unit tests worked with the 64 bit value but replaced with the correct Int128 value from MurmurHash3.